### DISCUSSION TITLE
Where can i read about writing and connecting my own third-party scripts?

### DISCUSSION THREAD

**Discussion by iWatchYouFromAfar** [nuxt/scripts#274] [Q&A]
I looked at [how the existing scripts are written](https://github.com/nuxt/scripts/tree/main/src/runtime/registry). But when i trying to write my own, i can't figure out how to skip `options` and actually run the function where `useRegistryScript` is called.

I added my script using [nuxt.hooks.](https://scripts.nuxt.com/docs/api/nuxt-hooks)

**Comment by harlan-zw** [comment]
Two paths:

1. **Simple** — just use `useScript()` directly. That's the supported public API and works for any script.
2. **Typed registry-style** — see [`useRegistryScript`](https://scripts.nuxt.com/docs/api/use-registry-script) plus the source of any registry under `packages/script/src/runtime/registry/` as reference.

Closing — open a docs issue if you'd like a step-by-step guide added 🙏
